src/components/Headers/DashboardHeader/DashboardHeader.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import Link from 'next/link'
+import { getServerSession } from 'next-auth'
+import { authOptions } from '@/lib/auth'
+import { redirect } from 'next/navigation'
+import { Role } from '@prisma/client'
+
+import { findUserByEmail } from '@/backend/user/user.service'
+import { AppRoutes } from '@/utils/routes'
+
+import { Container } from '@/components/Container/Container'
+import { Button } from '@/components/Button/Button'
+import { MyProfileButton } from '@/components/MyProfileButton/MyProfileButton'
+import logo from '@/assets/images/logo.png'
+
+import styles from './DashboardHeader.module.scss'
+
+const DashboardHeader = async () => {
+  const session = await getServerSession(authOptions)
+
+  const profile = session
```

Comment:
I wonder if we could shorten this code a bit?
For example:
`const profile = session && await findUserByEmail(session.user.email)`
`if(!profile || !profile?.roles.includes(Role.MODERATOR)) redirect(AppRoutes.home)`